app/gui2/vite.config.ts

```diff
@@ -94,13 +99,48 @@ export default defineConfig({
   },
 })
 
+let ydocServer: ChildProcessWithoutNullStreams | null
 function gatewayServer(): Plugin {
   return {
     name: 'gateway-server',
     configureServer(server) {
       if (server.httpServer == null) return
 
-      createGatewayServer(server.httpServer, undefined)
+      if (IS_POLYGLOT_YDOC_SERVER) {
+        const ydocServerJar = fileURLToPath(
+          new URL(
+            '../../lib/java/ydoc-server/target/ydoc-server-assembly-0.1.0-SNAPSHOT.jar',
```

Comment:
I am surprised this works at all! The `ydoc-server-assembly*.jar` is a "fat JAR" that contains all the classes - including classes from Graal.js. I thought the `TruffleLanguage` implementations must be provided by as Java modules to work fast. But everything seems to work fine even in this setup. CCing @Akirathan.

Ideally we don't use `-jar` option to Java launcher, but `--module-path`. The standard Engine launcher is going to use `--module-info` and it will use `graal-js` as its own module JAR.

Reply:
I was wrong. Apparently, the assembled jar still contains the `Graal.js` classes

Reply:
Yes, it does:
```
enso$ unzip -v lib/java/ydoc-server/target/ydoc-server-assembly-0.1.0-SNAPSHOT.jar | grep com.oracle.js | wc -l
155
```
otherwise there would be no JavaScript interpreter.